### DISCUSSION TITLE
How can vision AI can be used to detect wear on the inside of a tire? 🚗

### DISCUSSION THREAD

**Discussion by NLadi** [ultralytics/ultralytics#22202] [Show and tell]
Uneven wear can signal bigger issues like poor alignment, low pressure, or worn suspension parts. But these early signs are easy to miss.

In her latest blog, Abirami Vina explains how Vision AI and Ultralytics YOLO11 can detect subtle tire wear patterns with precision, helping drivers boost safety, extend tire life, and cut costs.

Learn more ➡️ https://www.ultralytics.com/blog/vision-ai-can-be-used-to-detect-wear-on-the-inside-of-a-tire

[![Tyres_thumbnail](https://github.com/user-attachments/assets/2cab8381-ecd8-45d3-817a-06aee9a55caf)](https://www.ultralytics.com/blog/vision-ai-can-be-used-to-detect-wear-on-the-inside-of-a-tire)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @NLadi, thank you for starting this discussion and sharing insights on using Vision AI with YOLO11 for tire wear detection 🚗✨ This is an automated response to help you get quick resources while an Ultralytics engineer will also assist soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Thanks @NLadi—nice overview; for anyone prototyping this, segment the inner-edge wear and train a YOLO11 segmentation model, then run inference on inspection videos; details in the blog [How Vision AI Can Identify Wear on the Inside of a Tire](https://www.ultralytics.com/blog/vision-ai-can-be-used-to-detect-wear-on-the-inside-of-a-tire), with setup guidance in [YOLO11 models](https://docs.ultralytics.com/models/yolo11/), [Segmentation task](https://docs.ultralytics.com/tasks/segment/), and [Train mode](https://docs.ultralytics.com/modes/train/). 

```python
from ultralytics import YOLO

# Train (masks for classes like 'wear_inner', 'wear_edge', etc.)
model = YOLO('yolo11n-seg.pt')
model.train(data='tire_wear_seg.yaml', epochs=50, imgsz=640)

# Inference on frames/videos
model.predict(source='inspections/', conf=0.25, save=True)
```
